### ISSUE TITLE
It doesn't work anymore on linux

### ISSUE THREAD

**Issue by dgiero** [fireice-uk/xmr-stak#53]
Hi, 
yesterday I cloned and compiled the miner on linux and it went all good. 
Today, on a different but identical pc, I have cloned and compiled the miner and it doesn't work, I made a lot of tries, the only way to make it works was to use the source code cloned yesterday.

The only thing I get is this:
```
-------------------------------------------------------------------
xmr-stak 2.0.0-predev mining software.
Based on CPU mining code by wolf9466 (heavily optimized by fireice_uk).
Brought to you by fireice_uk and psychocrypt under GPLv3.

Configurable dev donation level is set to 2.0 %

You can use following keys to display reports:
'h' - hashrate
'r' - results
'c' - connection
-------------------------------------------------------------------
[2017-10-18 14:42:43] : hwloc: memory pinned
```
then it stucks.

With the miner compiled from the old code (using the same config.txt and cpu.txt) I get:

```
-------------------------------------------------------------------
xmr-stak 2.0.0-predev mining software.
Based on CPU mining code by wolf9466 (heavily optimized by fireice_uk).
Brought to you by fireice_uk and psychocrypt under GPLv3.

Configurable dev donation level is set to 2.0 %

You can use following keys to display reports:
'h' - hashrate
'r' - results
'c' - connection
-------------------------------------------------------------------
[2017-10-18 14:49:22] : Starting double thread, affinity: 0.
[2017-10-18 14:49:22] : Starting single thread, affinity: 1.
[2017-10-18 14:49:22] : Starting double thread, affinity: 2.
[2017-10-18 14:49:22] : Starting single thread, affinity: 3.
[2017-10-18 14:49:22] : Connecting to pool xxxx:yyy ...
[2017-10-18 14:49:22] : Connected. Logging in...
[2017-10-18 14:49:22] : hwloc: memory pinned
[2017-10-18 14:49:22] : hwloc: memory pinned
[2017-10-18 14:49:22] : hwloc: memory pinned
[2017-10-18 14:49:22] : hwloc: memory pinned
[2017-10-18 14:49:22] : Difficulty changed. Now: 120001.
[2017-10-18 14:49:22] : New block detected.
```

I saw you made some change on the code in the last 24 hours, can you check the new code? 

Thanks

**Comment by psychocrypt** [comment]
Could you please clone it again and test it. I saw the same error in #44
and my tests finished without issues after I switched to the newest version.

**Comment by fireice-uk** [comment]
Thanks for reporting  - fixed in #54

**Comment by dgiero** [comment]
Its working! 
Thanks for your job

**Comment by psychocrypt** [comment]
solved with #54
